File: parse_days.js
const fs = require('fs');

const content = fs.readFileSync('data.js', 'utf8');
const regexDay = /day:\s*(\d+)/g;
const regexDateTitle = /date:\s*"([^"]+)",\s*title:\s*"([^"]+)"/g;

let days = [];
let matchDay;
while ((matchDay = regexDay.exec(content)) !== null) {
    let matchDateTitle = regexDateTitle.exec(content);
    if (matchDateTitle) {
        days.push(`Day ${matchDay[1]} | Date: ${matchDateTitle[1]} | Title: ${matchDateTitle[2]}`);
    }
}
console.log(days.join('\n'));

File: parse_days.py
import re

with open('data.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Match day, date, and title correctly using a tighter regex
pattern = r'day:\s*(\d+)[^\n]*\n\s*date:\s*"([^"]+)",\s*title:\s*"([^"]+)"'
matches = re.findall(pattern, content)

with open('days_list.txt', 'w', encoding='utf-8') as f:
    for d, date, title in matches:
        f.write(f"Day {d} | Date: {date} | Title: {title}\n")
